trigger.sh

```diff
@@ -0,0 +1,172 @@
+#
+# This script workflow will be like
+#       1. Parse arguments
+#              Note: if you will set module as KVM then no need to give IP address
+#       2. Create unique directory to store logs and data
+#	3. Check existance of virtualenv and if not exist then will create it.
+#       4. call prerequisite_changes.py file to make changes according to service
+#       5. deploy edgex
+#       6. start run test suite and save logs under unique directory
+#       7. shutdown edgex
+#
+# Note: After completion of test execution log report will be saved in working_dir/
+
+#!/bin/bash
+
+if [ $# -eq 0 ]
+then
+    echo "No arguments supplied, please provide data"
+    echo "Usage: sh trigger.sh --service functionalTest/V2-API/<any_service> --auth <username/password> --port <port_number> --module <module> --ip <ip_address>"
+    exit 1
+fi
+
+SHORT=s:,a:,p:,m:,i:
+LONG=service:,auth:,port:,module:,ip:
+OPTS=$(getopt -a -n weather --options $SHORT --longoptions $LONG -- "$@")
+
+eval set -- "$OPTS"
+
+while :
+do
+  case "$1" in
+    -s | --service )
+      service="$2"
+      shift 2
+      ;;
+    -a | --auth )
+      auth="$2"
+      shift 2
+      ;;
+    -p | --port)
+      port=$2
+      shift 2
+      ;;
+    -m | --module)
+      module=$2
+      shift 2
+      ;;
+    -i | --ip)
+      ip=$2
+      shift 2
+      ;;
+    --)
+      shift;
+      break
+      ;;
+    *)
+      echo "Unexpected option: $1"
+      ;;
+  esac
+done
+
+# parsing service name from functionalTest/V2-API/<core-service>
+test_service=`echo $service | awk -F"/" '{print $NF}'`
+service_flag=0
+
+service_list=(core-data
+              core-metadata
+              core-command
+              app-serice
+              support-notifications
+              support-schedular
+              system-agent
+             )
+
+for str in ${service_list[@]}; do
+  if [ $str == $test_service ]
+  then
+      service_flag=1
+  fi
+done
+
+if [ $service_flag == 0 ]
+then
+    echo "Please provice valid service"
+    exit 1
+fi
+
+#creating one timestamp based directory to store data
+data_dir=$(date +%y%m%d_%H%M%S)
+mkdir -p working_dir/${data_dir}
+
+# export some variables before starting execution
+export WORK_DIR=`pwd`
+export DATA_DIR=${WORK_DIR}/working_dir/${data_dir}
+export DATA_FILE=${DATA_DIR}/data.txt
+export SECURITY_SERVICE_NEEDED=true
+export DEPLOY_TYPE=manual
+export AUTH=${auth}
+export PORT=${port}
+export MODULE=${module}
+
+if [ $module == "STM" ] || [ $module == "stm" ]
+then
+    echo $ip > $DATA_FILE
+fi
+
+# creating VENV if it does not exist
+if [[ "$VIRTUAL_ENV" != " " ]]
+then
+    echo "=============="
+    echo "creating VENV"
+    echo "=============="
+    python3 -m venv .
+    source ./bin/activate
+    python3 -m pip install -U pip
+    pip install -U virtualenv
+fi
+
+# Install Robot Framework related Libraries
```

Comment:
Shouldn't we be using the requirements.txt file provided by this repo instead of doing each one individually?

Reply:
ok. I will create one separate requirements_eaton.txt file which includes robot framework related libraries. so there will be 2 requirements.txt files. one is our requiremets_eaton.txt and another one is for edgex-taf-common.

Reply:
I tested locally by removing requirements_eaton.txt file (also uninstalled same libraries from my local system) and simply installing requirements.txt from EdgeX-TAF and edgex-taf-common repo and it works. so will remove requirements_eaton.txt file.